config.py

```diff
@@ -125,6 +125,20 @@ def get_version():
     return AGENT_VERSION
 
 
+def _version_string_to_tuple(version_string):
+    '''Return a (X, Y, Z) version tuple from an 'X.Y.Z' version string'''
+    version_list = []
+    for elem in version_string.split('.'):
+        try:
+            elem_int = int(elem)
+        except ValueError:
+            log.warning("Unable to parse element '%s' of version string '%s'", elem, version_string)
```

Comment:
So your changes have actually highlighted something interesting. I actually think this should error out if we can't parse the version_string to ints. We know (or can assume with a very high degree of certainty) for a fact the `AGENT_VERSION` will be well formed, so that would be an indication of a bad string in the manifest. I would maybe favor raising an exception here and making sure `validate_sdk_check` returns `False`. The check will still run, as we provide a best-effort approach, and could be something valuable to log.